spec/integration/blocked_hotlinked_image_spec.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+describe "hotlinked image blocking" do
+  let(:hotlinked_url) { "http://example.com/images/2/2e/Longcat1.png" }
+  let(:onebox_url) { "http://example.com/onebox" }
+  let(:png) { Base64.decode64("R0lGODlhAQABALMAAAAAAIAAAACAAICAAAAAgIAAgACAgMDAwICAgP8AAAD/AP//AAAA//8A/wD//wBiZCH5BAEAAA8ALAAAAAABAAEAAAQC8EUAOw==") }
+
+  before do
+    SiteSetting.download_remote_images_to_local = false
+    stub_request(:get, hotlinked_url).to_return(body: png, headers: { "Content-Type" => "image/png" })
+    stub_image_size
+  end
+
+  it "normally allows hotlinked images" do
```

Comment:
I'm wondering if this has already been covered by specs else where.

Reply:
It's probably been covered as a side effect, yeah. But I'd prefer to comprehensively test this feature in its own spec file, so that future changes to other specs don't affect the coverage of this feature.